Content.Client/Disposal/UI/MailingUnitBoundUserInterface.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Content.Client.Disposal.Components;
+using Content.Client.Disposal.Systems;
+using Content.Shared.Disposal;
+using JetBrains.Annotations;
+using Robust.Client.GameObjects;
+using Robust.Client.UserInterface.Controls;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using static Content.Shared.Disposal.Components.SharedDisposalUnitComponent;
+
+namespace Content.Client.Disposal.UI
+{
+    /// <summary>
+    /// Initializes a <see cref="MailingUnitWindow"/> and updates it when new server messages are received.
+    /// </summary>
+    [UsedImplicitly]
+    public sealed class MailingUnitBoundUserInterface : BoundUserInterface
+    {
+        public MailingUnitWindow? Window;
+
+        public MailingUnitBoundUserInterface(ClientUserInterfaceComponent owner, object uiKey) : base(owner, uiKey)
+        {
+        }
+
+        private void ButtonPressed(UiButton button)
+        {
+            SendMessage(new UiButtonPressedMessage(button));
+            // If we get client-side power stuff then we can predict the button presses but for now we won't as it stuffs
+            // the pressure lerp up.
+        }
+
+        private void TargetSelected(ItemList.ItemListSelectedEventArgs args)
+        {
+            var item = args.ItemList[args.ItemIndex];
+            SendMessage(new TargetSelectedMessage(item.Text));
+        }
+
+        protected override void Open()
+        {
+            base.Open();
+
+            Window = new MailingUnitWindow();
+
+            Window.OpenCentered();
+            Window.OnClose += Close;
+
+            Window.Eject.OnPressed += _ => ButtonPressed(UiButton.Eject);
+            Window.Engage.OnPressed += _ => ButtonPressed(UiButton.Engage);
+            Window.Power.OnPressed += _ => ButtonPressed(UiButton.Power);
+
+            Window.TargetListContainer.OnItemSelected += TargetSelected;
+        }
+
+        protected override void UpdateState(BoundUserInterfaceState state)
+        {
+            base.UpdateState(state);
+
+            if (state is not MailingUnitBoundUserInterfaceState cast)
+            {
+                return;
+            }
+
+            Window?.UpdateState(cast);
+
+            // Kinda icky but we just want client to handle its own lerping and not flood bandwidth for it.
+            if (!IoCManager.Resolve<IEntityManager>().TryGetComponent(Owner.Owner, out DisposalUnitComponent? component)) return;
+
+            component.UiState = cast.DisposalState;
+            EntitySystem.Get<DisposalUnitSystem>().UpdateActive(component, true);
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            base.Dispose(disposing);
+
+            if (disposing)
+            {
+                Window?.Dispose();
+            }
+        }
```

Comment:
Just make it the same class as DisposalUnitBoundUserInterface?

Reply:
I really don't want to spent any more time on something so nitpicky when fixing it is more involved than using an abstract state. The state technically is already abstract, that's why almost all bound user interface classes contain the *state is not ...* boilerplate.

I have an Idea how I can easily merge the two bound user interface classes but if that gets rejected I will just revert it and the code stays a bit duplicated

Reply:
Like fixing the smaller nitpicks above would've only taken me a few minutes but I've spent most of my free evening addressing this now even if the solution *seems* simple. (I don't have much free time because of work)

Reply:
fyi, removing copy pasted code isnt a nitpick